symbolic-debuginfo/src/elf.rs

```diff
@@ -537,7 +537,13 @@ impl<'data> ElfObject<'data> {
 
     /// Determines whether this object contains embedded source.
     pub fn has_sources(&self) -> bool {
-        false
+        // TODO: can we determine if we have sources without opening a debug session?
```

Comment:
this is indeed a bit unfortunate, in particular as you iterate over *all* the files in the worst case.

Reply:
Yes. The possible alternative here would be to check if the DWARFv5 file info table includes a `DW_LNCT_LLVM_source` entry. This'd still require opening a DebugSession, but doesn't require iterating over each file.

Just having the `DW_LNCT_LLVM_source` entry included in the file records doesn't mean that we've actually got sources available though, and this approach would stop working in DWARF v6 where file info records become modular, so I'd stick with iterating over each file :slightly_smiling_face: